LONDON | EMILIANO URUENA | Project-TV-Show Halyna | Sprint2 - Level 200

Emilianouz opened the pull request:
LONDON | EMILIANO URUENA | Project-TV-Show Halyna | Sprint2 - Level 200

Self checklist

- [X] I have committed my files one by one, on purpose, and for a reason
- [X] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [X] I have tested my changes
- [X] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [X] My changes meet the [requirements](./README.md) of this task

## Changes: 

I implemented a search-by-input feature that filters episodes in real-time as the user types, and added a dropdown episode selector that allows users to view individual episodes based on formatted episode codes. I also dynamically created and styled a navigation bar to hold these controls without modifying the original HTML. The counter text label was rebuild including the amount of selected episodes. These changes improved the interactivity and structure of the page, using clean JavaScript logic and utility functions to maintain readability and consistency.

## Comments about our projects

1. How is it different?
Your implementation differs from mine in a few noticeable ways. One key structural difference is that you created a semantic <main> container to wrap the episode content, which improves readability and accessibility. Additionally, you consistently used arrow functions (=>) throughout your code, which is a more modern syntax and helps with concise function expressions.

2. What do you prefer about your implementation?
One area where I feel confident about my approach is in the formatting of episode numbers. I used a dedicated formatEpisodeNumber() function that ensures consistent padding and formatting for season and episode codes (e.g., S01E02). This made it easier to reuse and maintain across the project.

3. What do you prefer about their implementation?
Your implementation had a polished and thoughtful feel to it. I especially appreciated how you added a footer section—it gave the layout a complete and professional structure. Your use of semantic HTML, such as choosing <article> tags instead of just <section>, reflects a good understanding of best practices.

I also noticed that your CSS was more detailed and deliberate. The styling was well-structured, visually appealing, and gave each component clear boundaries and personality.

4. What did you learn that you didn't know before?
I learned about the ternary operator from your code, which was completely new to me. Seeing how you used it for conditional rendering helped me understand how to write more concise logic. It's a useful tool that I can now apply in my own work going forward.

Emilianouz committed:
Search by input and Episode selector implemented. Adding nav. Changes on HTML, JS and Styles.

Emilianouz committed:
Refactor using state and render function.

halyna-k commented on index.html:
```diff
@@ -10,6 +10,8 @@
   </head>
 
   <body>
+    <nav id="nav">
```
@Emilianouz The nav element is meant for page or section navigation. For search/filter functionality within the same page, better use section, div, or form to keep the markup semantic and accessible.

halyna-k commented on script.js:
```diff
@@ -74,17 +81,93 @@ function footer() {
+  let selectSearch = document.createElement("select");
+  selectSearch.name = "select-search";
+  selectSearch.id = "select-search";
+  state.selectSearch = "selectSearch"; // added to save the reference on the state
```
@Emilianouz The assignment should store the DOM element reference, not a string literal: state.selectSearch = selectSearch;

halyna-k commented on script.js:
```diff
@@ -74,17 +81,93 @@ function footer() {
+  optionAll.textContent = "Show All"
+  selectSearch.appendChild(optionAll);
+
+  listEpisodes = allEpisodes.map((episode) => {
```
@Emilianouz The variable listEpisodes is declared without const, let, or var, making it an implicit global variable.

halyna-k commented on style.css:
```diff
@@ -23,9 +23,9 @@ body {
+h3 {
+  font-size: 18px;
+  padding: inherit;
 }
```
@Emilianouz Also here some suggestion: 
Use relative units for scalability & avoid inheriting padding blindly

Hard-coding font-size: 18px breaks user zoom / root-font scaling; prefer 1.125rem.
padding: inherit propagates all four sides from the parent, which is rarely what you want on headings and can lead to unpredictable spacing. Consider explicit padding: 0 (or remove it entirely) and control spacing via margins instead.

-h3 {
-  font-size: 18px;
-  padding: inherit;
+h3 {
+  font-size: 1.125rem; /* 18 px @ 16 px root */
+  /* padding intentionally unset – rely on margins for spacing */
 }

halyna-k commented on style.css:
```diff
@@ -104,6 +104,27 @@ footer a {
+  font-weight: normal;
+  padding-left: 50px;
+  margin-left: 40px;
+}
```
@Emilianouz Just let you know: Consider responsive spacing & accessibility for search inputs
padding-left: 50px and margin-left: 40px are large fixed pixel values; on mobile widths they may cause horizontal scroll or cramped layouts. Use relative units (rem, %) or a media query.

halyna-k commented on style.css:
```diff
@@ -104,6 +104,27 @@ footer a {
+  background: var(--main-bg-color);
+  opacity: 0.75;
+  box-shadow: 0 -2px 5px rgba(0,0,0,0.1);
+}
```
@Emilianouz nav is missing position: fixed; align-items: left is invalid CSS

left/top have no effect without a positioning context, so the bar won’t stick to the top as intended.
align-items: left isn’t a valid value (allowed values: flex-start, center, etc.) – this will be ignored by browsers.

 nav {
   ...
+  position: fixed;   /* ensure bar stays at the top */
   ...
-  align-items: left;
+  align-items: center; /* or flex-start, depending on desired cross-axis alignment */
   ...
-  box-shadow: 0 -2px 5px rgba(0,0,0,0.1);
+  box-shadow: 0 2px 5px rgba(0,0,0,0.1); /* drop shadow below the bar */
 }

halyna-k commented on script.js:
```diff
@@ -35,18 +45,15 @@ function createEpisodeContainer() {
-
 // Function to create the episode card
-function createEpisodeCard(episode) {
+function createEpisodeCard(episode) { // It will be this the render function?
```
@Emilianouz The comment appears to be leftover development notes and should be removed for cleaner code.

halyna-k commented on script.js:
```diff
@@ -74,17 +81,93 @@ function footer() {
+
+function createNav() {
+  const navElem = document.querySelector("nav");
+  ///const allEpisodes = getAllEpisodes();
```
@Emilianouz Commented out code should be removed to keep the codebase clean.

halyna-k commented on script.js:
```diff
@@ -74,17 +81,93 @@ function footer() {
+// function after update input-search
+function searchByInput(){
+  const query = document.getElementById("input-search").value;
+  //const allEpisodes = getAllEpisodes();/** */
```
@Emilianouz Same here. Commented out code should be removed to keep the codebase clean.

halyna-k commented on script.js:
```diff
@@ -74,17 +81,93 @@ function footer() {
+//create render
+  // clean
+  // filter
+  // build
```
@Emilianouz Here as well. These appear to be development notes that should be removed from the final code.

Emilianouz committed:
Changes were implemented according to the suggestions.

Emilianouz committed:
Updated design and layout structure: replaced IDs, applied relative units for scalability.
Refactored HTML by replacing <nav> with <div> to support search/filter functionality.
Updated scripts to reflect DOM structure changes and maintain reference integrity.
